src/raytracer/xmlreader.m

```diff
@@ -63,114 +65,149 @@
-                    CADOutput(indexCADOutput,14) = MaterialLibrary.PrimaryKey(iterateMaterials);
-                    switch2=1;
+        %% Extracting the vertices information of the triangles
+
+        sizeTriangle=size(triangles);
+        CADOutputTemp = [];
+        for iterateTriangles=1:sizeTriangle(1)
+%             indexCADOutput=countRows;
+            if isfield(s.amf,'Attributes')
+                switch s.amf.Attributes.unit
+                    case 'millimeter'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*0.001;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*0.001;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*0.001;   
+                    case 'inch'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*0.0254;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*0.0254;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*0.0254;   
+                    case 'foot'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*0.3048;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*0.3048;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*0.3048;   
+                    case 'micrometer'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*1e-6;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*1e-6;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*1e-6;   
+                    case 'meter'
+                        v1 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v1',sizeVolume)*1;   
+                        v2 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v2',sizeVolume)*1;   
+                        v3 = getTriangleVertex(stemp,iterateVolume,iterateTriangles,'v3',sizeVolume)*1;                   
+                    otherwise
+                        error('xmlreader does not support this unit.');
+                end
+            else
+               error('Length unit is missing in the xml/amf file. Add <amf  unit="?" in Line 1>');
+            end
+            % Calculating the plane equation of triangles
+
+            vector1 = v2 - v3;
+            vector2 = -(v2 - v1);
+            % Multiply with -1 for 'example.xml', 'sphere.xml','material_prism.xml'
+            normal = cross(vector2,vector1) * (1-(2*IndoorSwitch));
+            normal = round(normal/norm(normal),4);
+            vector3 = v2;
+            % for box. remove for others
```

Comment:
is this comment still needed?

Reply:
Not sure. Need to confirm with Anuraag.

Reply:
it's not even clear what it means. if you keep it, please clarify it, otherwise i would remove it